File: services/api/internal/models/history.go
package models

type PlayerHistoryItem struct {
	GameID   int64   `json:"game_id"`
	Title    string  `json:"title"`
	PlayedAt string  `json:"played_at"`
	Score    *int    `json:"score,omitempty"`
	Status   *string `json:"status,omitempty"`
}

type PlayerHistoryPagination struct {
	Page  int `json:"page"`
	Limit int `json:"limit"`
	Total int `json:"total"`
}

type PlayerHistoryResponse struct {
	Data       []PlayerHistoryItem     `json:"data"`
	Pagination PlayerHistoryPagination `json:"pagination"`
}

File: apps/web/src/lib/api/history.ts
import { api } from "$lib/api/client";
import { getToken } from "$lib/auth/playerAuth";

export type PlayerHistoryItem = {
    game_id: number;
    title: string;
    played_at: string;
    score?: number;
    status?: string;
};

export type PlayerHistoryPagination = {
    page: number;
    limit: number;
    total: number;
};

export type PlayerHistoryResponse = {
    data: PlayerHistoryItem[];
    pagination: PlayerHistoryPagination;
};

export type PlayerHistoryParams = {
    page?: number;
    limit?: number;
};

function buildQuery(params: PlayerHistoryParams = {}): string {
    const q = new URLSearchParams();

    if (typeof params.page === "number") {
        q.set("page", String(params.page));
    }
    if (typeof params.limit === "number") {
        q.set("limit", String(params.limit));
    }

    const qs = q.toString();
    return qs ? `?${qs}` : "";
}

export function getPlayerHistory(params: PlayerHistoryParams = {}): Promise<PlayerHistoryResponse> {
    const qs = buildQuery(params);
    const token = getToken();

    return api.get<PlayerHistoryResponse>(`/player/history${qs}`, {
        token: token ?? undefined,
    });
}
